Sources/HttpPipeline/NIO.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+import NIO
+import NIOHTTP1
+import Optics
+import Prelude
+
+public func run(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, Prelude.Unit, Data>,
+  on port: Int = 8080
+  ) {
+
+  do {
+    let reuseAddrOpt = ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR)
+    let group = MultiThreadedEventLoopGroup(numThreads: System.coreCount)
+    let bootstrap = ServerBootstrap(group: group)
+      .serverChannelOption(ChannelOptions.backlog, value: 256)
+      .serverChannelOption(reuseAddrOpt, value: 1)
+      .childChannelInitializer { channel in
+        channel.pipeline.configureHTTPServerPipeline()
+          .then { channel.pipeline.add(handler: HTTPResponseCompressor()) }
```

Comment:
This enables gzipped responses. Not sure if it's worth making this configurable or not or if it should be disabled in dev. Seems fine?